### DISCUSSION TITLE
Grpc-web for communication between backend services

### DISCUSSION THREAD

**Discussion by calamas** [grpc/grpc-web#1455] [Ecosystem support]
Hello everyone. I am working on a project that uses gRPC for communication between services in a Kubernetes environment. In order to properly balance the traffic between pods, since gRPC uses HTTP/2, we had to install Linkerd, and everything is working fine now.

The issue is that my company finds Linkerd to be expensive and wants to replace gRPC with simple HTTP/1 calls to avoid this cost.

My question is: Is it possible, efficient, and secure to use gRPC-Web for communication between backend services? Has anyone successfully done this?

**Comment by sampajano** [comment]
@calamas Hi! grpc-web maintainer here. One thing to note is that this grpc-web library is built in particular with browser in mind. So maybe it's possible that you could make it work in other environments, but you'd be on your own in terms of support :)
